### PR TITLE
docs(installation): add docker installation video tutorials and refactor the page structure

### PR DESCRIPTION
**Pull request by cyn-liu**
Signed-off-by: Cynthia Liu <cynthia.liu@autocore.ai>

## Description

Related issue: https://github.com/autowarefoundation/autoware-documentation/issues/270

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/installation/autoware/docker-installation.md [hunk 1]
```diff
@@ -4,150 +4,23 @@
...
+## Docker installation for quick start
 
-1. Clone `autowarefoundation/autoware` and move to the directory.
+[docker installation for quick start](./docker-installation-prebuilt.md)
 
-   ```bash
-   git clone https://github.com/autowarefoundation/autoware.git
-   cd autoware
-   ```
+[Reference video tutorials](https://drive.google.com/file/d/1P0hCJ8jQHRMQ1xGINDjqn4MQpiMOMVGd/view?usp=share_link)
```
Comment (xmfcx) [inline docs/installation/autoware/docker-installation.md:17]:
```suggestion
![type:video](https://drive.google.com/file/d/1P0hCJ8jQHRMQ1xGINDjqn4MQpiMOMVGd/view?usp=share_link)
```

We've added https://pypi.org/project/mkdocs-video/ with https://github.com/autowarefoundation/autoware-github-actions/pull/202

Now it should support `type:video` syntax. Let's try it.

File: docs/installation/autoware/docker-installation.md [hunk 2]
```diff
@@ -4,150 +4,23 @@
...
+## Docker installation for quick start
 
-1. Clone `autowarefoundation/autoware` and move to the directory.
+[docker installation for quick start](./docker-installation-prebuilt.md)
 
-   ```bash
-   git clone https://github.com/autowarefoundation/autoware.git
-   cd autoware
-   ```
+![type:video](https://drive.google.com/file/d/1P0hCJ8jQHRMQ1xGINDjqn4MQpiMOMVGd/view?usp=share_link)
```
Comment (xmfcx) [inline docs/installation/autoware/docker-installation.md:17]:
```suggestion
[Reference video tutorials](https://drive.google.com/file/d/1P0hCJ8jQHRMQ1xGINDjqn4MQpiMOMVGd/view?usp=share_link)
![type:video](https://www.youtube.com/embed/LXb3EKWsInQ)
```
Hmm the video tag didn't work on the google drive link.

Screenshot from 2023-01-07 01-53-40

I want to test if it works at all with their example in https://pypi.org/project/mkdocs-video/

**Comment by cyn-liu** [comment]
Hi @xmfcx 
You suggestion is to put the video tutorial on youtube and  then link the video in the form of `` ?

File: docs/installation/autoware/docker-installation.md [hunk 3]
```diff
@@ -4,150 +4,24 @@
...
 
-1. Clone `autowarefoundation/autoware` and move to the directory.
+[docker installation for quick start](./docker-installation-prebuilt.md)
 
-   ```bash
-   git clone https://github.com/autowarefoundation/autoware.git
-   cd autoware
-   ```
+[Reference video tutorials](https://drive.google.com/file/d/1P0hCJ8jQHRMQ1xGINDjqn4MQpiMOMVGd/view?usp=share_link)
+![type:video](https://www.youtube.com/embed/LXb3EKWsInQ)
```
Comment (kenji-miyake) [inline docs/installation/autoware/docker-installation.md:18]: Updating `mkdocs.yaml` is required.

Comment (kenji-miyake) [reply docs/installation/autoware/docker-installation.md:18]: Also, the link is the same as the example, not Autoware's video.

**Comment by kenji-miyake** [comment]
If possible, it's better to split this PR into two.
1. Refactor the page structure.
2. Add a reference video link.

**Comment by cyn-liu** [comment]
> If possible, it's better to split this PR into two.
> 
>     1. Refactor the page structure.
> 
>     2. Add a reference video link.

@kenji-miyake thank you very much for your reply. 
What do you think of changing the title instead  dividing this PR into two.

**Comment by kenji-miyake** [comment]
@cyn-liu Yeah, that also works for me!

File: docs/installation/autoware/docker-installation.md (hunk 3 above)
Comment (cyn-liu) [reply docs/installation/autoware/docker-installation.md]: I use this feature update my video links, but I found these links cant not be opened.
How to solve this problem?
图片

Comment (kenji-miyake) [reply docs/installation/autoware/docker-installation.md]: You mean this?
image

Comment (kenji-miyake) [reply docs/installation/autoware/docker-installation.md]: Can be fixed by this change.
image

```diff
diff --git a/docs/installation/autoware/docker-installation.md b/docs/installation/autoware/docker-installation.md
index 48d5f9df2..eae63229f 100644
--- a/docs/installation/autoware/docker-installation.md
+++ b/docs/installation/autoware/docker-installation.md
@@ -14,13 +14,13 @@ Here are two ways to install Autoware by docker:
 
 [docker installation for quick start](./docker-installation-prebuilt.md)
 
-![Reference video tutorials](https://youtu.be/3KUhEFkEbI8)
+![type:video](https://youtube.com/embed/3KUhEFkEbI8)
 
 ## Docker installation for development
 
 [docker installation for development](./docker-installation-devel.md)
 
-![Reference video tutorials](https://youtu.be/UrSF-VwncGQ)
+![type:video](https://youtube.com/embed/UrSF-VwncGQ)
 
 ## Troubleshooting
 
diff --git a/mkdocs.yaml b/mkdocs.yaml
index 8f1c36060..5c0ab8214 100644
--- a/mkdocs.yaml
+++ b/mkdocs.yaml
@@ -53,10 +53,9 @@ plugins:
         - ^(?!(.*/)?assets/).*\.(?!(.*\.)?md|(.*\.)?svg|(.*\.)?png|(.*\.)?jpg).*$
         - ^(.*/)?[^.]*$
   - macros
+  - mkdocs-video
   - same-dir
   - search
-  - mkdocs-video:
-      mark: "Reference video tutorials"
 
 markdown_extensions:
   - abbr
```

Comment (kenji-miyake) [reply docs/installation/autoware/docker-installation.md]: By the way, have you enabled this checkbox? If you do, I can push my fixes.
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

Comment (cyn-liu) [reply docs/installation/autoware/docker-installation.md]: Yes, you can push your fixes. thank you.

image

Comment (kenji-miyake) [reply docs/installation/autoware/docker-installation.md]: I believe now it's fixed.

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**